guvnor-ala/guvnor-ala-wildfly-provider/src/main/java/org/guvnor/ala/wildfly/access/WildflyClient.java

```diff
@@ -0,0 +1,296 @@
+/*
+ * Copyright 2016 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.guvnor.ala.wildfly.access;
+
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.IOException;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreType;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import org.apache.http.HttpEntity;
+import org.apache.http.HttpResponse;
+import org.apache.http.auth.AuthScope;
+import org.apache.http.auth.UsernamePasswordCredentials;
+import org.apache.http.client.CredentialsProvider;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.entity.mime.MultipartEntityBuilder;
+import org.apache.http.entity.mime.content.ByteArrayBody;
+import org.apache.http.entity.mime.content.FileBody;
+import org.apache.http.impl.client.BasicCredentialsProvider;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.jboss.dmr.ModelNode;
+
+import java.util.Date;
+import static java.util.logging.Level.*;
+import static java.util.logging.Logger.*;
+import static org.apache.http.entity.ContentType.APPLICATION_JSON;
+import static org.apache.http.entity.mime.HttpMultipartMode.*;
+import static org.apache.http.entity.mime.MultipartEntityBuilder.create;
+import static org.apache.http.impl.client.HttpClients.*;
+import static org.apache.http.entity.ContentType.create;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.util.EntityUtils;
+
+/**
+ * Based on: https://github.com/heiko-braun/http-upload
+ */
+@JsonIgnoreType
+public class WildflyClient {
+
+    private String providerName;
+    private String user;
+    private String password;
+    private String host;
+    private int managementPort;
+
+    public WildflyClient() {
+    }
+
+    public WildflyClient( String providerName, String user, String password, String host, int managementPort ) {
+        this.providerName = providerName;
+        this.user = user;
+        this.password = password;
+        this.host = host;
+        this.managementPort = managementPort;
+    }
+
+    /*
+     * return the HTTP status from deploying the app or -1 if there is an exception which is logged.
```

Comment:
I see that you are throwing an exception in this method now (thanks!) but now the comment is inaccurate.